src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/interceptor/AuthorisationInterceptor.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.blobrouter.reconciliation.interceptor;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.hmcts.reform.blobrouter.exceptions.InvalidApiKeyException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static com.microsoft.applicationinsights.boot.dependencies.google.common.net.HttpHeaders.AUTHORIZATION;
+
+@Component
+public class AuthorisationInterceptor implements HandlerInterceptor {
```

Comment:
will this interceptor not be executed for all the controller endpoints ? Do we really need an interceptor ?

Reply:
It'll be executed only for the reconciliation endpoint. I didn't add the config file in this PR as it will lead to more changes in the tests.It'll be like this:
` registry.addInterceptor(authorisationInterceptor)
            .addPathPatterns("/reform-scan/reconciliation-report/*");`

Reply:
you are implementing HandlerInterceptor and it is beaned up so this will be executed for all endpoints

Reply:
nice. it's been a while i've fiddled with interceptors. i guess there are multiple ways to do this. there are 2 or 3 different abstractions of `HandlerInerceptor` which would also suffice what's needed. config is way cleaner/simpler ^

Reply:
may be not but that config should go in single PR as it is interceptor config

Reply:
ok, I'll include config changes to this PR

Reply:
@nitinprabhu added config changes and controller changes.

Reply:
perhaps instead of creating an interceptor we could just add another parameter to the controller method?
```
@RequestHeader("Authorization") String authHeader,
```

Reply:
ok. I'll delete the interceptor and move the validation to the controller

Reply:
@rkondratowicz I already added the `Authorization` header. Are you suggesting to include another parameter? what would it be used for?